users/api/metrics.go

```diff
@@ -100,3 +100,85 @@ func (ms *metricsMiddleware) SendPasswordReset(ctx context.Context, host, email,
+	defer func(begin time.Time) {
+		ms.counter.With("method", "create_group").Add(1)
+		ms.latency.With("method", "create_group").Observe(time.Since(begin).Seconds())
+	}(time.Now())
+
+	return ms.svc.CreateGroup(ctx, token, group)
+}
+
+func (ms *metricsMiddleware) ListGroups(ctx context.Context, token, id string, offset, limit uint64, meta users.Metadata) (users.GroupPage, error) {
+	defer func(begin time.Time) {
+		ms.counter.With("method", "list_groups").Add(1)
+		ms.latency.With("method", "list_groups").Observe(time.Since(begin).Seconds())
+	}(time.Now())
+
+	return ms.svc.ListGroups(ctx, token, id, offset, limit, meta)
+}
+
+func (ms *metricsMiddleware) ListGroupUsers(ctx context.Context, token, id string, offset, limit uint64, meta users.Metadata) (users.UserPage, error) {
+	defer func(begin time.Time) {
+		ms.counter.With("method", "list_users_in_groups").Add(1)
+		ms.latency.With("method", "list_users_in_groups").Observe(time.Since(begin).Seconds())
+	}(time.Now())
+
+	return ms.svc.ListGroupUsers(ctx, token, id, offset, limit, meta)
+}
+
+func (ms *metricsMiddleware) RemoveGroup(ctx context.Context, token, id string) error {
+	defer func(begin time.Time) {
+		ms.counter.With("method", "remove_group").Add(1)
+		ms.latency.With("method", "remove_group").Observe(time.Since(begin).Seconds())
+	}(time.Now())
+
+	return ms.svc.RemoveGroup(ctx, token, id)
+}
+
+func (ms *metricsMiddleware) UpdateGroup(ctx context.Context, token string, group users.Group) error {
+	defer func(begin time.Time) {
+		ms.counter.With("method", "update_group").Add(1)
+		ms.latency.With("method", "update_group").Observe(time.Since(begin).Seconds())
+	}(time.Now())
+
+	return ms.svc.UpdateGroup(ctx, token, group)
+}
+
+func (ms *metricsMiddleware) ViewGroup(ctx context.Context, token, name string) (users.Group, error) {
+
+	defer func(begin time.Time) {
+		ms.counter.With("method", "view_group").Add(1)
+		ms.latency.With("method", "view_group").Observe(time.Since(begin).Seconds())
+	}(time.Now())
+
+	return ms.svc.ViewGroup(ctx, token, name)
+}
+
+func (ms *metricsMiddleware) AssignUserToGroup(ctx context.Context, token, userID, groupID string) error {
+	defer func(begin time.Time) {
+		ms.counter.With("method", "assign_user_to_group").Add(1)
+		ms.latency.With("method", "assign_user_to_group").Observe(time.Since(begin).Seconds())
+	}(time.Now())
+
+	return ms.svc.AssignUserToGroup(ctx, token, userID, groupID)
+}
+
+func (ms *metricsMiddleware) UnassignUserFromGroup(ctx context.Context, token, userID, groupID string) error {
+	defer func(begin time.Time) {
+		ms.counter.With("method", "remove_user_from_group").Add(1)
+		ms.latency.With("method", "remove_user_from_group").Observe(time.Since(begin).Seconds())
+	}(time.Now())
+
+	return ms.svc.UnassignUserFromGroup(ctx, token, userID, groupID)
+}
+
+func (ms *metricsMiddleware) ListUserGroups(ctx context.Context, token, id string, offset, limit uint64, meta users.Metadata) (users.GroupPage, error) {
```

Comment:
`ListGroups()`

Reply:
this lists groups that user is assigned to